.github/workflows/ci.yml

```diff
@@ -0,0 +1,68 @@
+name: Node.js CI with PostgreSQL
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: tbd
+          POSTGRES_DB: mockpfi
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    strategy:
+      matrix:
+        node-version: ['20.4.0']
+
+    env:
+      ENV: production
+      LOG_LEVEL: info
+      HOST: http://localhost:9000
+      PORT: 9000
+      #SEC_DB_HOST: postgres # Use the service name for the host
+      #SEC_DB_PORT: 5432
+      #SEC_DB_USER: postgres
+      #SEC_DB_PASSWORD: tbd
+      #SEC_DB_NAME: mockpfi
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Install dependencies
+      run: npm install
+    - name: Install dbmate
+      run: |
+        curl -fsSL https://github.com/amacneil/dbmate/releases/download/v1.12.1/dbmate-linux-amd64 -o dbmate
+        chmod +x dbmate
+        sudo mv dbmate /usr/local/bin      
+    - name: Run migration scripts
+      run: |
+        ./db/scripts/migrate
```

Comment:
thoughts on updating [`db/scripts/migrate`]() to use dbmate docker container to run migrations?

e.g.
```bash
echo "Running migrations for $DATABASE_URL"
docker run --rm --detach \
  --network=host \
  --env DATABASE_URL=$DATABASE_URL \
  --name tbdex-pfi-dbmate \
  --volume "$(pwd)/db:/db" \
  --entrypoint /bin/sh ghcr.io/amacneil/dbmate -c 'dbmate wait && dbmate up'
```

may have to tweak ^

this way we can use the same scripts to do things locally and for CI. then can just run

```yml
- name: Run migration scripts
  run: ./db/scripts/migrate
```

Reply:
similar to above - that won't work in actions as no docker daemon available once inside an action